### PR TITLE
Py3 compatibility

### PR DESCRIPTION
**Pull request by jhkennedy**
A variety of changes to (mostly) ensure python 3 compatibility and some style fixes. The only thing I haven't done in that regard is to make python 2 behave as if it's python 3 with:

```
from __future__ import print_function, division, unicode_literals, absolute_import
```

See: https://python-future.org/compatible_idioms.html

Major changes in this PR are:
* Fix (my) bad import of urlparse
* Some essential style fixes:
  * Remove `import *`s
  * Remove unused imports
  * Remove all trailing semicolons
  * Don't shadow built-in python functions (fixes some, not all)
  * Use `is` for comparison to `None`

### CODE REVIEW

File: src/resample_geotiff.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def resample_geotiff(geotiff, width, outFormat, outFile, use_nn = False):
   if outFormat.upper() == 'GEOTIFF':
     gdal.Translate(outFile, raster, resampleAlg=resampleMethod, width=width)
   elif outFormat.upper() == 'JPEG' or outFormat.upper() == 'JPG':
-    if colorTable == None:
+    if colorTable is not None:
```
Comment (talogan) [inline src/resample_geotiff.py]: We should be checking for colorTable is None here.

File: src/resample_geotiff.py [hunk 2]
```diff
@@ -90,7 +90,7 @@ def resample_geotiff(geotiff, width, outFormat, outFile, use_nn = False):
     tmpFile = outFile.replace(orgExt, tmpExt)
     rgbFile = None
     if bandCount == 1:
-      if colorTable == None:
+      if colorTable is not None:
```
Comment (talogan) [inline src/resample_geotiff.py]: Same for this one - colorTable is None.

File: src/resample_geotiff.py (hunk 1 above)
Comment (jhkennedy) [reply src/resample_geotiff.py]: Ack! That was sloppy of me -- fixed now

File: src/resample_geotiff.py (hunk 2 above)
Comment (jhkennedy) [reply src/resample_geotiff.py]: ditto

**Comment by jhkennedy** [comment]
@talogan I'm out today, so you can go ahead and merge if you approve!

**Pull request merged**